# Correlating small-cap pharmaceutical drug companies' stock prices with the FDA drug approval process
CS 198-134: Introduction to Quantitative Finance Final Project

Presented by Cyprian Zander, Enmanual Felix Pena, Jeet Sandhu, Pranay Rajpaul, Shangdian Han

## Abstract

_Write out details about pharmaceutical drug companies, cite some sources, state hypothesis. Here are some small details that may be helpful:_

Discuss pharmaceutical company stocks typically having high volatility.

##### FDA Drug Approval Process

https://www.drugwatch.com/fda/approval-process/
https://www.fda.gov/patients/drug-development-process/step-3-clinical-research
<br />Terms: NDA (New Drug Application)

**Incentive for pharmaceutical company**
<br />A new patent is good for 20 years from the date the application is filed in the U.S.

**Expedited review process**
<br />Prescription Drug User Fee Act (PDUFA) allows the FDA to collect fees from companies to expedite the drug-approval process.
<br />Standard Review: Drug approval within 10 months.
<br />Priority Review: Drug approval within 6 months.

**Reasons for denying approval**
<br />Unexpected safety issues, manufacturing issues or a failure to prove a drug’s effectiveness.

**The drug approval process**
<br />Basic safety of drug is tested through chemical tests and animal tests.
<br />Clinical trials consist of 3 phases.
1. Phase 1
    - Length: Several months
    - Participants: 20 to 100 healhy volunteers / people with disease
    - Selectivity: 70% of drugs pass
    
    - Intentions
    - How drug interacts with human body
    - Adjusted dosing based on animal data
    - Find out how much body can tolerate + acute side effects
    - How to best administer the drug to limit risks and maximize benefits
    
    
2. Phase 2
    - Length: Several months to 2 years 
    - Participants: Several hundred people with disease
    - Selectivity: 33% of drugs pass
    
    - Intentions
    - Not large enough to show whether drug will be beneficial
    - Additional safety data
    - Research questions + research methods + design P3 research protocols


3. Phase 3
    - Length: 1 to 4 years
    - Participants: 300 to 3000
    - Selectivity: 25-30%
    
    - Intentions
    - Test if it offers a treatment benefit to a specific population
    - "Pivotal studies"
    - Provides most safety data - reveals long term or rare side effects

Passing of phase 3 - FDA Approval given. Drug released to market.

4. Phase 4
    - Several thousand volunteers
    
    - Carried out once drug approved by FDA in Post Market Safety Monitoring

**Notes**
<br />Medical Devices have Class I, II, III based on how invasive they are - Pharmaceutical companies are allowed to announce devices before approval process.

## Method

Testing this hypothesis can be broken into the following steps:

1. Collecting small cap pharmaceutical company tickers: find all indexed pharmaceutical stocks, filter by market cap

2. Collect each stock's news history: filter for FDA drug approvals and rejections

3. Collect price change data: 1mo at 1hr intervals of stock data following news

4. Analysis: Compare stock performance following news vs. performance of same time frame drawing from a random sample of ticker price history (_or another method of testing the hypothesis?_)

### Importing Libraries

In [2]:
import yfinance as yf
import pandas as pd
import numpy as np
from tqdm import tqdm

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

### 1. Collecting small cap pharmaceutical company tickers

Nasdaq's stock screener (https://www.nasdaq.com/market-activity/stocks/screener) enables filtering of stocks by exchange, market cap, analyst rating, sector, region, and country. We collect tickers from NYSE, NASDAQ, and AMEX by filtering out stocks with market cap > $10B, as these stocks would no longer be considered "small cap", thus falling outside of the parameters of our hypothesis and searching by sector: "Healthcare".

The prepared CSV file now contains 1200 observations, but these also need to be filtered by industry ("Biotechnology: Pharmaceutical Preparations"), as this industry focusses on pharmaceutical drug preparation.

In [15]:
healthcare_stocks = pd.read_csv("nasdaq_screener_1683316923031.csv")

pharmaceutical_preparations = healthcare_stocks.loc[healthcare_stocks["Industry"] == "Biotechnology: Pharmaceutical Preparations"] # Filter by Industry
pharmaceutical_preparations.head()
print("pharmaceutical_preparations contains", len(pharmaceutical_preparations.index), "tickers.")


Symbol                                      Name Last Sale  Net Change  \
0   AADI         Aadi Bioscience Inc. Common Stock     $7.94      0.1600   
3   ABEO     Abeona Therapeutics Inc. Common Stock     $3.32      0.0600   
5   ABOS  Acumen Pharmaceuticals Inc. Common Stock   $5.1675      1.1175   
6   ABSI            Absci Corporation Common Stock     $1.40      0.0900   
8   ABVC          ABVC BioPharma Inc. Common Stock     $0.80      0.0500   

  % Change   Market Cap        Country  IPO Year   Volume       Sector  \
0   2.057%  194029701.0  United States       NaN    46995  Health Care   
3    1.84%   59198980.0  United States       NaN   104342  Health Care   
5  27.593%  211997008.0  United States    2021.0  1167028  Health Care   
6    6.87%  129352873.0  United States    2021.0   519172  Health Care   
8   6.667%   26464592.0  United States       NaN    36071  Health Care   

                                     Industry  
0  Biotechnology: Pharmaceutical Preparations  
3  Biotechnology: Pharmaceutical Preparations  
5  Biotechnology: Pharmaceutical Preparations  
6  Biotechnology: Pharmaceutical Preparations  
8  Biotechnology: Pharmaceutical Preparations

pharmaceutical_preparations contains 619 tickers.



### 2. Collect each stock's news history

Having collected all 619 small-cap pharmaceutical preparation stock tickers, 

## Analysis and Conclusion

_Write text on how to perform analysis_

## Bibliography

_tbd_